gomock/matchers.go

```diff
@@ -97,6 +97,18 @@ func (anyMatcher) String() string {
 	return "is anything"
 }
 
+type fnMatcher struct {
```

Comment:
What are your thoughts on making this type just a redefinition of the func type since it doesn't really need to be a struct?
```
type fnMatcher fn(any) bool

func (f fnMatcher) Matches(x any) bool {
    return f(x)
}
```